user_sync/config.py

```diff
@@ -254,8 +254,14 @@ def load_from_yaml(self, file_path):
         '''
         :type file_path: str
         '''        
-        with open(file_path, 'r', 1) as input_file:
-            return yaml.load(input_file)
+        try:
+            with open(file_path, 'r', 1) as input_file:
+                return yaml.load(input_file)
+        except yaml.error.MarkedYAMLError as e:
+            # if a parser error occured while loading the configuration file,
+            # swallow the error and re-raise it as a configuration loader
+            # exception
+            raise user_sync.error.AssertionException('Error loading configuration file: %s' % e)
```

Comment:
Please make clear that it was a YAML syntax error, not some sort of general error.
Also please catch any file-open exceptions unless there's some other code that already ensures the file is readable.